### PR TITLE
[SAGE-619] make list and table rows clickable

### PR DESCRIPTION
**Pull request by QuintonJason**
## Description

- [x] update the click target of the primary link in the catalog item and table components
- [x] updated the z-index of all nested interactive elements to sit above the expanded hitbox

#### Catalog Item (Rails)

|  Before  |  After  |
|--------|--------|
|catalogItemRailsBefore|catalogItemRails|

#### Table (Rails)
|  Before  |  After  |
|--------|--------|
|tableRailsBefore|tableRails|

#### Table (React)
|  Before  |  After  |
|--------|--------|
|tableReactBefore|tableReact|

## Testing in `sage-lib`

Verify that the entire row is the interactive trigger instead of only the primary link:
- [Catalog Item](http://localhost:4000/pages/component/catalog_item?tab=preview) (Rails)
- [Table (Rails)](http://localhost:4000/pages/component/table?remove=&tab=preview)
- [Table (React)](http://localhost:4110/?path=/story/sage-table--table-with-rich-content)

## Testing in `kajabi-products`

1. (**HIGH**) Changes the click target of Catalog Items and Tables to be the entire row.

## Related

Closes [SAGE-619](https://kajabi.atlassian.net/browse/SAGE-619)

### CODE REVIEW

File: packages/sage-react/lib/themes/next/Table/sample-data/domains.jsx [hunk 1]
```diff
@@ -1,11 +1,20 @@
...
 import { Label } from '../../Label';
+import { Link } from '../../Link';
 import { SageTokens } from '../../configs';
 
 export const domains = [
   {
-    domain: 'www.daydreamsurfshop.mykajabi.com',
+    domain: (
+      <Link
+        className="sage-table-cell__link--primary"
```
Comment (philschanely) [inline packages/sage-react/lib/themes/next/Table/sample-data/domains.jsx:11]: See above regarding adding this in `SageClassnames` for React too?

File: docs/app/views/examples/components/themes/next/table/_preview.html.erb [hunk 2]
```diff
@@ -224,6 +224,112 @@ sample_table = {
...
+              icon: { name: 'trash', style: 'only' },
+              style: 'danger',
+              attributes: {
+                href: "?remove=#delete",
+                "data-js-tooltip": "Delete",
+              }
+            })
+          }
+        </div>
+      )
```
Comment (philschanely) [inline docs/app/views/examples/components/themes/next/table/_preview.html.erb:282]: Consider this variation for cleaner documentation:

```suggestion
      link_to("User", "#", class: "sage-link sage-link--remove-underline sage-table-cell__link--primary"),
      "$20,323.30 USD",
      {
        value: "Sed posuere consectetur",
        html_attributes: {
          class: "sage-table-cell--truncate"
        }
      },
      sage_component(SageBadge, {
        value: "published",
        color: "published",
      }),
      sage_component(SageButtonGroup, {
        content: (
          sage_component(SageButton, {
            subtle: true,
            value: 'Edit',
            icon: { name: 'pen', style: 'only' },
            style: 'primary',
            attributes: {
              href: "?edit=#edit",
              "data-js-tooltip": "Edit",
            }
          }) << sage_component(SageButton, {
            subtle: true,
            value: 'Delete',
            icon: { name: 'trash', style: 'only' },
            style: 'danger',
            attributes: {
              href: "?remove=#delete",
              "data-js-tooltip": "Delete",
            }
          })
        ),
        gap: :xs
      }),
```

File: docs/app/views/examples/components/themes/next/table/_preview.html.erb [hunk 3]
```diff
@@ -224,6 +224,112 @@ sample_table = {
...
+    "Malesuada",
+    "Amet Cras",
+    "Malesuada",
+    "Amet Cras",
+  ],
+  selectable: true,
+  rows: [
+    [
+      %(
+        #{link_to("User", "#", class: "sage-link sage-link--remove-underline sage-table-cell__link--primary")}
```
Comment (philschanely) [inline docs/app/views/examples/components/themes/next/table/_preview.html.erb:240]: What do you think about adding `sage-table-cell__link--primary` in SageClassnames?

Comment (philschanely) [reply docs/app/views/examples/components/themes/next/table/_preview.html.erb:240]: Also do you think a SageLink could be used here instead of `link_to`?

**Review by philschanely — APPROVED** [review]
Fabulous work and solid approach! Just some documentation suggestions here but 👍 otherwise! 🏆

**Review by monkeypox8 — APPROVED** [review]
👍 LGTM, working great

**Comment by QuintonJason** [comment]
Closed in favor of https://github.com/Kajabi/sage-lib/pull/1532

**Pull request closed without merging**